tests/test_atom_mapping.py

```diff
@@ -953,3 +965,190 @@ def test_get_cores_and_diagnostics():
         assert (
             diagnostics.total_nodes_visited >= diagnostics.core_size
         )  # must visit at least one node per atom pair in core
+
```

Comment:
A few more basic tests seem useful here (unless I'm missing something), e.g.

1. do we get the same result with `initial_mapping=None`, `initial_mapping=[]`, etc.?
2. given an arbitrary initial mapping, does the result always contain the initial mapping?
3. if we run without an initial mapping and then again passing a subset of the generated mapping as the initial mapping, is the resulting mapping equivalent to the first one we generated?
4. if the initial mapping isn't consistent with the specified filters, do we generate an error?

Reply:
Added tests in https://github.com/proteneer/timemachine/pull/1254/commits/c8bf1aa22edbe57e00bb9b3a58d5ad59bc0cf137

4. has some nuance, depending on the parameter then either no core can be found and returns the original core or an exception is raised. Could throw an exception if the core returned is the identity map provided, which seems like it might be cleaner.

Reply:
Thinking about it some more, don't want to throw an exception if the core returned is the identity map. That would make this painful to use in practice, where the new parameters may not generate a larger core.